core/events/abstract.js

```diff
@@ -71,7 +71,7 @@ Blockly.Events.Abstract = function(elem) {
    */
   this.recordUndo = undefined;
 
-  if (elem instanceof Blockly.Block) {
+  if (elem instanceof Blockly.Block || elem instanceof Blockly.WorkspaceComment) {
```

Comment:
If it's a workspace comment save the id in `this.commentId` instead.  That way if we have comments on blocks (with similar events) we can store the block id correctly.